src/main/java/org/jboss/tools/intellij/openshift/utils/odo/Odo.java

```diff
@@ -58,16 +58,16 @@
 
     List<URL> listURLs(String project, String application, String context, String component) throws IOException;
 
-    ComponentInfo getComponentInfo(String project, String application, String component) throws IOException;
+    ComponentInfo getComponentInfo(String project, String application, String component, ComponentKind kind) throws IOException;
 
     void createURL(String project, String application, String context, String component, String name, Integer port,
                    boolean secure) throws IOException;
 
     void deleteURL(String project, String application, String context, String component, String name) throws IOException;
 
-    void undeployComponent(String project, String application, String context, String component) throws IOException;
+    void undeployComponent(String project, String application, String context, String component, ComponentKind kind) throws IOException;
 
-    void deleteComponent(String project, String application, String context, String component, boolean undeploy) throws IOException;
+    void deleteComponent(String project, String application, String context, String component, ComponentKind kind) throws IOException;
```

Comment:
the removal of the undeploy parameter is not clear to me, needs to discuss it

Reply:
undeploy paramter is never used in the odoCli current implementation. 
tested with various states (pushed, no context, not pushed ) and both types (s2i/devfile )